src/frontend/src/icp/components/transactions/IcTransaction.svelte

```diff
@@ -1,74 +1,51 @@
 	import { BigNumber } from '@ethersproject/bignumber';
-	import { type ComponentType } from 'svelte';
 	import IcTransactionLabel from '$icp/components/transactions/IcTransactionLabel.svelte';
 	import type { IcTransactionType, IcTransactionUi } from '$icp/types/ic-transaction';
-	import IconConvert from '$lib/components/icons/IconConvert.svelte';
-	import IconConvertFrom from '$lib/components/icons/IconConvertFrom.svelte';
-	import IconConvertTo from '$lib/components/icons/IconConvertTo.svelte';
-	import IconReceive from '$lib/components/icons/IconReceive.svelte';
-	import IconSend from '$lib/components/icons/IconSend.svelte';
-	import TransactionStatusComponent from '$lib/components/transactions/TransactionStatus.svelte';
-	import Amount from '$lib/components/ui/Amount.svelte';
-	import Card from '$lib/components/ui/Card.svelte';
-	import RoundedIcon from '$lib/components/ui/RoundedIcon.svelte';
+	import Transaction from '$lib/components/transactions/Transaction.svelte';
+	import { NANO_SECONDS_IN_SECOND } from '$lib/constants/app.constants';
 	import { modalStore } from '$lib/stores/modal.store';
 	import type { TransactionStatus } from '$lib/types/transaction';
-	import { formatNanosecondsToDate } from '$lib/utils/format.utils';
 
 	export let transaction: IcTransactionUi;
 
 	let type: IcTransactionType;
 	let transactionTypeLabel: string | undefined;
 	let value: bigint | undefined;
-	let timestamp: bigint | undefined;
+	let timestampNanoseconds: bigint | undefined;
 	let incoming: boolean | undefined;
 
-	$: ({ type, typeLabel: transactionTypeLabel, value, timestamp, incoming } = transaction);
+	$: ({
+		type,
+		typeLabel: transactionTypeLabel,
+		value,
+		timestamp: timestampNanoseconds,
+		incoming
+	} = transaction);
 
 	let pending = false;
 	$: pending = transaction?.status === 'pending';
 
 	let status: TransactionStatus;
 	$: status = pending ? 'pending' : 'confirmed';
 
-	let icon: ComponentType;
-	$: icon =
-		['burn', 'approve', 'mint'].includes(type) && pending
-			? IconConvert
-			: ['burn', 'approve'].includes(type)
-				? IconConvertTo
-				: type === 'mint'
-					? IconConvertFrom
-					: type === 'send'
-						? IconSend
-						: IconReceive;
-
 	let amount: bigint | undefined;
 	$: amount = !incoming && nonNullish(value) ? value * -1n : value;
-</script>
-
-<button on:click={() => modalStore.openIcTransaction(transaction)} class="block w-full border-0">
-	<Card>
-		<span class="inline-block first-letter:capitalize"
-			><IcTransactionLabel label={transactionTypeLabel} fallback={type} /></span
-		>
 
-		<RoundedIcon slot="icon" {icon} />
-
-		<svelte:fragment slot="amount">
-			{#if nonNullish(amount)}
-				<Amount amount={BigNumber.from(amount)} />
-			{/if}
-		</svelte:fragment>
-
-		<svelte:fragment slot="description">
-			{#if nonNullish(timestamp)}
-				{formatNanosecondsToDate(timestamp)}
-			{/if}
+	let timestamp: number | undefined;
+	$: timestamp = nonNullish(timestampNanoseconds)
```

Comment:
i found a few utils in NNS dapp in module `date.utils`, but nothing regarding NANOSECONDS TO SECONDS. Do you remember maybe the usage?

Reply:
Maybe I mixed up, thanks for the feedback.

We have a `formatNanosecondsToDate` which uses `NANO_SECONDS_IN_MILLISECOND` for number. Why not using this constant?

And I was thinking, what do you think of extended `Transaction` to path a property `Date` for Timestamp? Feels a bit less prone to issue if consumer can have timestamp in various formats.

Reply:
> Why not using this constant?

for this one, we need to adapt IC timestamp (nanoseconds) to the generic one (seconds) that is the same as ETH and BTC.

> extended Transaction to path a property Date for Timestamp

this will be done in another PR, since we need to group transactions by date, as per new requirements